src/Discord/Parts/Part.php

```diff
@@ -178,12 +178,28 @@ public function fetch(): ExtendedPromiseInterface
     public function fill(array $attributes): void
     {
         foreach ($attributes as $key => $value) {
-            if (in_array($key, $this->fillable)) {
+            if (in_array($key, $this::getFillableAttributes())) {
                 $this->setAttribute($key, $value);
             }
         }
     }
 
+	/**
+     * Returns the fillable attributes.
+     *
+     * @return array
+     */
+    public static function getFillableAttributes($context = '')
+	{
+		$fillable = array();
+		foreach (self::$fillable as $attr) {
```

Comment:
`self::$fillable` will always reference `Part::$fillable` even if the function is called on a child class, you can use the `static::` keyword instead to reference the class that the function is called on (`static::$fillable`).

Reply:
I did some testing and confirmed that this is indeed the case; Unless the function itself is overridden in the child classes, self::$fillable will call the parent method if that's where the function is being defined. Each individual Part would need its own getFillableAttribute function if the self keyword was to be used.